smac/optimizer/acquisition/maximizer.py

```diff
@@ -455,17 +455,25 @@ def _do_search(
                 if not active[i]:
                     continue
                 if obtain_n[i] == 0 or improved[i]:
-                    obtain_n[i] = 2
+                    obtain_n[i] = self.vectorization_min_obtain
                 else:
                     obtain_n[i] = obtain_n[i] * 2
                     obtain_n[i] = min(obtain_n[i], self.vectorization_max_obtain)
                 if new_neighborhood[i]:
                     if not improved[i] and n_no_plateau_walk[i] < self.n_steps_plateau_walk:
-                        if len(neighbors_w_equal_acq[i]) != 0:
+                        if len(neighbors_w_equal_acq[i]) > 0:
```

Comment:
I am not sure if we really need this. Assuming that we have lots of hyperparameters, as each time we only sample 8 neighbors, it is quite likely that all these 8 neighbors are all worse than the current one while the acquisition function values can be further reduced with the other neighbors